phenotrex/transforms/annotation.py

```diff
@@ -7,29 +8,50 @@
 from concurrent.futures import ProcessPoolExecutor
 
 import torch
+from deepnog.inference import load_nn, predict
+from deepnog.io import create_df, get_weights_path
+from deepnog.utils import set_device
+from deepnog.dataset import ProteinDataset, ProteinIterator
 from Bio.SeqIO import SeqRecord, parse
-from deepnog.deepnog import load_nn, predict, set_device, create_df
-from deepnog.dataset import ProteinDataset
 from tqdm.auto import tqdm
 
 from phenotrex.io.flat import load_fasta_file
 from phenotrex.structure.records import GenotypeRecord
 
-
 PRODIGAL_BIN_PATH = resource_filename('phenotrex', 'bin/prodigal')
-DEEPNOG_WEIGHTS_PATH = resource_filename('deepnog', 'parameters/eggNOG5/2/deepencoding.pth')
 DEEPNOG_ARCH = 'deepencoding'
-EGGNOGDB_VERS = '5.0'
+DEEPNOG_VALID_DBS = {
+    'eggNOG5',
+}
+DEEPNOG_VALID_TAX_LEVELS = {
+    1,
+    2,
+}
+
+
+class PreloadedProteinIterator(ProteinIterator):
+    """Hack ProteinDataset to load from list directly."""
+    def __init__(self, protein_list: List[SeqRecord], aa_vocab, format):
+        self.iterator = (x for x in protein_list)
```

Comment:
Is that equivalent?
```suggestion
        self.iterator = protein_list
```

Reply:
Yeah, should be, true

Reply:
Nope, deepNOG complained about not being passed an iterator (it tried to call next() on the list)

Reply:
Ah yes, the comprehension gives a generator, not a tuple, and so does not allocate memory twice. But since it's called "self.iterator", how about the following:
```suggestion
        self.iterator = iter(protein_list)
```

Reply:
well technically a generator is an iterator too, but I concede the point that this is cleaner ^^

Reply:
You're right. Both is fine, no hard feelings